tests/tests/history_api_tests.cpp

```diff
@@ -604,5 +604,75 @@ BOOST_AUTO_TEST_CASE(get_account_history_operations) {
       throw;
    }
 }
+//new test case for increasing the limit based on the config file
+BOOST_AUTO_TEST_CASE(setting_max_operation_limit_get_account_history_operations) {
+  try {
+    graphene::app::history_api hist_api(app);
+
+    //account_id_type() do 3 ops
+    create_bitasset("CNY", account_id_type());
+    create_account("sam");
+    create_account("alice");
+
+    generate_block();
+    fc::usleep(fc::milliseconds(2000));
+
+    int asset_create_op_id = operation::tag<asset_create_operation>::value;
+    int account_create_op_id = operation::tag<account_create_operation>::value;
+    int transfer_op_id = operation::tag<graphene::chain::transfer_operation>::value;
+
+    //account_id_type() did 1 asset_create op
+    vector<operation_history_object> histories = hist_api.get_account_history_operations(
+	"committee-account", asset_create_op_id, operation_history_id_type(), operation_history_id_type(), 200);
+    BOOST_CHECK_EQUAL(histories.size(), 1);
+    BOOST_CHECK_EQUAL(histories[0].id.instance(), 0);
+    BOOST_CHECK_EQUAL(histories[0].op.which(), asset_create_op_id);
+
+    //account_id_type() did 2 account_create ops
+    histories = hist_api.get_account_history_operations(
+	"committee-account", account_create_op_id, operation_history_id_type(), operation_history_id_type(), 200);
+    BOOST_CHECK_EQUAL(histories.size(), 2);
+    BOOST_CHECK_EQUAL(histories[0].op.which(), account_create_op_id);
+
+    // No asset_create op larger than id1
+    histories = hist_api.get_account_history_operations(
+	"committee-account", asset_create_op_id, operation_history_id_type(), operation_history_id_type(1), 200);
+    BOOST_CHECK_EQUAL(histories.size(), 0);
+
+    // Limit 1 returns 1 result
+    histories = hist_api.get_account_history_operations(
+	"committee-account", account_create_op_id, operation_history_id_type(),operation_history_id_type(), 1);
+    BOOST_CHECK_EQUAL(histories.size(), 1);
+    BOOST_CHECK_EQUAL(histories[0].op.which(), account_create_op_id);
+
+    // alice has 1 op
+    histories = hist_api.get_account_history_operations(
+	"alice", account_create_op_id, operation_history_id_type(),operation_history_id_type(), 200);
+    BOOST_CHECK_EQUAL(histories.size(), 1);
+    BOOST_CHECK_EQUAL(histories[0].op.which(), account_create_op_id);
+
+    // create a bunch of accounts
+    for(int i = 0; i < 126; ++i)
+    {
+      std::string acct_name = "mytempacct" + std::to_string(i);
+      create_account(acct_name);
+    }
+    generate_block();
+
+    // history is set to limit transactions to 125 (see database_fixture.hpp)
+    // so asking for more should only return 125 (and not throw exception,
+    // see https://github.com/bitshares/bitshares-core/issues/1490
+    histories = hist_api.get_account_history_operations(
+	"committee-account", account_create_op_id, operation_history_id_type(), operation_history_id_type(), 125);
+    BOOST_CHECK_EQUAL(histories.size(), 125);
+    if (histories.size() > 0)
```

Comment:
Can use BOOST_REQUIRE in the line above then you don't need to check here.

Reply:
Done updated the code

Reply:
Ok, but the `histories.size() > 0` check is still there...